docs/database/includes/postgresql-app-host.md

```diff
@@ -44,13 +44,71 @@ var exampleProject = builder.AddProject<Projects.ExampleProject>()
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `docker.io/library/postgres` image, it creates a new PostgreSQL server instance on your local machine. A reference to your PostgreSQL server and database instance (the `postgresdb` variable) are used to add a dependency to the `ExampleProject`.
+
+When adding a database resource to the app model, the database is created if it doesn't already exist. The creation of the database relies on the [app host eventing APIs](../../app-host/eventing.md), specifically <xref:Aspire.Hosting.ApplicationModel.ResourceReadyEvent>. In other words, when the `postgres` resource is _ready_, the event is raised and the database resource is created.
+
+The PostgreSQL server resource includes default credentials with a `username` of `"postgres"` and randomly generated `password` using the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter*> method.
 
 The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `"messaging"`. For more information, see [Container resource lifecycle](../../fundamentals/app-host-overview.md#container-resource-lifecycle).
 
 > [!TIP]
 > If you'd rather connect to an existing PostgreSQL server, call <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> instead. For more information, see [Reference existing resources](../../fundamentals/app-host-overview.md#reference-existing-resources).
 
+### Add PostgreSQL resource with database scripts
+
+By default, when you add a <xref:Aspire.Hosting.ApplicationModel.PostgresDatabaseResource>, it relies on the following script to create the database:
+
+```sql
+CREATE DATABASE "<QUOTED_DATABASE_NAME>"
+```
+
+<!-- TODO: Use xref here when available
+
+To alter the default script, chain a call to the <xref:Aspire.Hosting.PostgresBuilderExtensions.WithCreationScript*> method on the database resource builder:
+
+-->
+
+To alter the default script, chain a call to the `WithCreationScript` method on the database resource builder:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var postgres = builder.AddPostgres("postgres");
+
+var databaseName = "app_db";
+var creationScript = $$"""
+    -- Create the database
+    CREATE DATABASE {{databaseName}};
+
+    -- Connect to the database
+    \c {{databaseName}}
```

Comment:
This is actually not supported. The SQL Server integration does have support for `GO` but we decided not to support `\c` in Postgres for now. The main reason being that `\c` allows you to switch databases, and the first one is on the `postgres` default db, so we would have to create a new connection, it's a little more involved than the `GO` statement in SQL as it remains on the same db.

I feel like we should still do it, but I already have many other things to finish for 9.2.

Reply:
Did you try to run this code and it worked? I could see a reason why it would, the fact that you use a db named `postgres`.

Reply:
The `databaseName` variable is `app_db`. I didn't test this out though, I was trying to have symmetry between the SQL creation scripts. This is a good callout though; I've adjusted this and added a note about the `\c` limitation.